internal/controller/services/gateway/resources/networkpolicy.yaml

```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+  labels:
+    app: {{ .Name }}
+    app.kubernetes.io/component: authentication
+    app.kubernetes.io/managed-by: opendatahub-operator
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ .Name }}
+  policyTypes:
+    - Ingress
+  ingress:
+    # Allow traffic from gateway/envoy pods for authentication
+    - from:
+        - podSelector:
+            matchLabels:
+              gateway.networking.k8s.io/gateway-name: {{ .GatewayName }}
+          namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: {{ .Namespace }}
+      ports:
+        - protocol: TCP
+          port: {{ .HTTPSPort }}
+    
+    # Allow metrics collection from OpenShift monitoring (port {{ .MetricsPort }})
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: openshift-monitoring
```

Comment:
Is this too permissive? Any pod in `openshift-monitoring` namespace can access metrics, not just Prometheus.

Reply:
I think it is fine, this namespace is only for admins, restricting  too much could lead to some issues when something changes.Also I followed a similar network policy pattern in ODH https://github.com/opendatahub-io/opendatahub-operator/blob/9d4a072803d2920fe4f8994d019186e6c9716cc3/config/monitoring/networkpolicy/operator/operator.yaml#L4